src/main/scala/fi/oph/kouta/images/s3Service.scala

```diff
@@ -44,39 +53,39 @@ class S3Service(private val s3Client: AmazonS3) extends Logging {
   lazy val publicUrl: Regex = s"${config.imageBucketPublicUrl}/(.*)".r
   lazy val tempUrl: Regex = s"${config.imageBucketPublicUrl}/temp/(.*)".r
 
-  def storeTempImage(contentType: ContentType, imageData: Array[Byte]): String = {
```

Comment:
Yleisesti vielä jäin miettimään tätä toteutusta... Meillä nyt on kokoraja noille kuville ja siinä mielessä tästä ei ehkä tule ongelmia, että koko kuva luetaan kerralla muistiin. Yleensä tällaiset kuvien uploadit olisi kuitenkin parempi tehdä niin, että se httpServletRequestin stream streamataan vaan suoraan eteen päin s3:een eikä sitä siis lueta muistiin missään välissä kokonaan. Tämä siksi, ettei muisti täyty siinä vaiheessa, jos joku päättää uploadata ison kuvan ja/tai monta käyttäjää uploadaa monta isohkoa kuvaa yhtä aikaa samalle instanssille. Muuttaisin ehkä tätä toteutusta vielä niin, että koko kuvaa ei lueta kerralla muistiin ellei ole ihan pakko.

Reply:
Kuvat luetaan muistiin sen takia, että niiden (pikseli-) koko ja tyyppi voidaan tarkistaa. Stream voidaan lukea kuitenkin vain kerran ja jos se streamataan S3:seen, sille ei voi tehdä käytännössä mitään välissä. Jos noita tarkistuksia ei olisi, se tietenkin streamattaisiin suoraan.

Reply:
Joo. Tiedän kyllä, koska olen itse koodannut vaikka kuinka monta kertaa vastaavia palikoita.

Reply:
Hyvää on se, että onneksi se maksimikoko tarkistetaan heti ensimmäisenä! August oli sitä mieltä, että korjataan tätä sitten, jos sen kanssa oikeasti tulee ongelmia.